activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -333,6 +333,22 @@ def foreign_key(table_name, options = {}) # :nodoc:
         foreign_keys[table_name] = options
       end
 
+      def remove_foreign_key(options_or_to_table) # :nodoc:
+        if options_or_to_table.is_a?(Hash) and options_or_to_table.has_key?(:name)
+          foreign_keys.delete_if do |table_name, options|
+            options[:name].to_s == options_or_to_table[:name].to_s
+          end
+        elsif options_or_to_table.is_a?(Hash) and options_or_to_table.has_key?(:column)
+          foreign_keys.delete_if do |table_name, options|
+            options[:column].to_s == options_or_to_table[:column].to_s
+          end
+        elsif options_or_to_table.is_a?(String) or options_or_to_table.is_a?(Symbol)
+          foreign_keys.delete options_or_to_table.to_s
+        else
+          raise ArgumentError, "remove_foreign_key needs either a table name or a foreign key options hash with :name or :column"
+        end
+      end
+
```

Comment:
I believe the rails style guide suggests using `&&` and `||` and braces on `.delete()` ?

Also I think the logic can be simplified a bit along the lines of
```
def remove_foreign_key(oott)
  if oott.is_a?(Hash)
    name_or_column = oott.has_key?(:name) ? :name : :column
    foreign_keys.delete_if do |_, options|
      options[name_or_column].to_s == oott[name_or_column].to_s
    end
  elsif oott.is_a?(String) || oott.is_a?(Symbol)
    foreign_keys.delete(oott.to_s)
  else
    raise ArgumentError, "remove_foreign_key needs either a table name or a foreign key options hash with :name or :column"
  end    
end
```

(At least I think this version is a bit easier to read)